content_sync/tasks.py

```diff
@@ -392,3 +393,48 @@ def check_incomplete_publish_build_statuses():
             log.exception(
                 "Error updating publishing status for website %s", website.name
             )
+
+
+@app.task(bind=True)
+@single_task(
+    timeout=settings.UPDATE_WEBSITES_IN_ROOT_WEBSITE_FREQUENCY, raise_block=False
+)
+def update_websites_in_root_website(self):
+    """
+    Get all websites published to draft / live at least once, and for each one:
+        - Create or update a WebsiteContent object of type website in the website denoted by settings.ROOT_WEBSITE_NAME
+        - Publish the content to draft / live branch of the root website in the Git backend
+    """
+    dirpath = "content/websites"
+    root_website = Website.objects.get(name=settings.ROOT_WEBSITE_NAME)
+    # Get all sites, minus any sites that have never been successfully published
+    sites = Website.objects.exclude(
+        Q(**{"draft_publish_date__isnull": True}) & Q(**{"publish_date__isnull": True})
+    )
+    sites = sites.exclude(Q(url_path__isnull=True))
+    # Exclude the root website
+    sites = sites.exclude(name=settings.ROOT_WEBSITE_NAME)
+    for site in sites:
+        site_metadata = WebsiteContent.objects.get(
+            website=site, type="sitemetadata"
+        ).metadata
+        # We want this content to show up in lists, but not render pages
+        site_metadata["_build"] = {"list": True, "render": False}
+        # Set the content to draft if the site has been unpublished
+        site_metadata["draft"] = True if site.unpublish_status is not None else False
+        # Carry over url_path for proper linking
+        site_metadata["url_path"] = site.url_path
+        WebsiteContent.objects.update_or_create(
+            website=root_website,
+            type="website",
+            title=site.title,
+            dirpath=dirpath,
+            filename=site.short_id,
+            is_page_content=True,
+            defaults={"title": site.title, "metadata": site_metadata},
+        )
+    website_content = WebsiteContent.objects.filter(
+        website=root_website, type="website"
+    )
+    github_api = github.GithubApiWrapper(website=root_website)
```

Comment:
Ideally it would be better to do this instead, in order to avoid having code explicitly linked to github as much as possible:

```python
backend = get_sync_backend(website)
backend.sync_all_content_to_backend(query_set=website_content)
```
And you would need to change the function definition for `BaseSyncBackend` and `GithubBackend` to include that optional parameter and pass it on to `GithubApiWrapper`.

However, the default way the github wrapper here works is to commit to main, then merge from main to draft or publish branch when published, and you need to commit directly to either draft or publish branch, which will need to be different all the time, right?  You could add an additional parameter for that.  Maybe something like:

BaseSyncBackend:
```python
def sync_all_content_to_backend(
  self, 
  query_set: Optional[WebsiteContentQuerySet] = None, 
  **kwargs
):
```

GithubBackend:
```python
def sync_all_content_to_backend(
  self, 
  query_set: Optional[WebsiteContentQuerySet] = None, 
  branch=setttings.GIT_BRANCH_MAIN
):
```

in code:
```
kwargs = {"branch": <preview/release>}
backend.sync_all_content_to_backend(query_set=website_content, **kwargs)
```

and then pass those on to the GithubApiWrapper, though that's where things might get real messy.  I think there's some assumptions in that code that everything gets committed to main.  When there's a standard publish triggered, main gets merged to draft or live.  The `WebsiteContent` objects autogenerated in this PR *might* get picked up with that and included in the resources to commit to main and then merged to preview/release, which would not be good - so might need to always include a queryset that excludes those resources (by `type="website"`?) when committing to git anywhere outside of this specific task, to ensure they stay out of the main branch.

My hunch is that as long as these new resources are saved only to the preview/release branches and not main, there won't be any conflicts when merging from main, but I'm not positive, probably will require some testing to find out for sure.

Reply:
Will also need to remove these from git if a site is unpublished

Reply:
Thanks for the suggestions. I switched from directly calling the Github backend to calling the generic one. I experimented with pushing the content directly up to `preview` / `live`, and the solution ends up being very messy. I wasn't even able to quite get it to work, I'm not sure exactly why but it had something to do with the `ContentSyncState` not evaluating in a way that the content would be included to commit, even after clearing said `ContentSyncState` objects. I'm somewhat stuck on this, but I'm thinking it might be a better idea to create a new branch in each content repo for syncing these changes and always push the content up there. In this case, the branch would be called something like `websites`. Then, we would add a new optional argument to the `publish_website` function to specify a source branch. I'm thinking it would go something like this:

 - Hard reset `websites` branch to `preview`
 - Commit the "website" content directly to the `websites` branch based on the current contents of the DB
 - Merge `websites` branch to `preview`
 - Repeat the process for `live`

In theory, setting `draft: true` on websites that have only been published in draft should allow us to exclude those websites from the index if `--buildDrafts` is not enabled.

Reply:
I don't recall exactly how the draft toggle works, but If we use `draft: <true/false>` to differentiate between `website` content objects that have been published to draft vs live, can we just push to the main branch for this type of content like all the others?  

`draft: true`: include in the draft list of courses only
`draft: false`: include in the draft and live list of courses